### DISCUSSION TITLE
the response time of the voice changer keeps increesing and after a while stops working

### DISCUSSION THREAD

**Discussion by veryangryapple** [w-okada/voice-changer#1275] [Q&A] [answered]
https://www.youtube.com/watch?v=yxG5pBrnzbU

I tried many things fallowed a lot of tutorials on how to make it faster so I made the file onnx then lowered the chunk to 1 and extra to minimum but it still refused to work normally so at this point I have doubts that its not because of my system 

graphics card: NVIDIA GeForce RTX 3060(12G)
CPU: AMD Ryzen 5 5600 6-Core Processor     3.50 GHz

cpu prob isnt supposed to work in the first place since I downloaded the NVIDIA version but wanted to add just in case

any and all help would be greatly appreciated

**Comment by AnnoyingFox** [comment]
Setting the chunk size to 1 won't make it work correctly. Set it to 128 (0.3 seconds) and set the extra to something around 8000.

I am not sure what are you trying to do in regards to making it "faster" but you can't get rid of the delay if you want it to sound properly. It's not a hardware limitation, the software needs to know what you are saying to convert it properly. And the more that you say (bigger chunk and delay) the better the conversion.

**Reply by veryangryapple** [reply]
![resim_2024-07-16_072634934](https://github.com/user-attachments/assets/ec07a291-eda9-4e79-81c2-6c7e74f6c1cc)

 I am assuming that vol stands for volume buf stands for buffer and res stands for response time 
 
 my "res" is keep increasing no matter what I use (chunk/extra) the greater res gets the more delayed my voice gets after a while (when "res" reaches a certain point) the application stops working until I stop it and start it once again but same thing happens again
 
 I hope I did make it more understandable this time I am sorry as English is not my native language

**Comment by AnnoyingFox** [comment, marked as answer]
Try rmvpe instead of rmvpe_onnx and try with a pth file instead of an onnx one. I would also recommend not using any of the vb cable products and instead use virtual audio cable (VAC) as vb cable have a lot of issues. It may not affect the res directly but can affect the quality. Make sure to update your nvidia drivers to the latest version as well.
Huge response times usually means that the system is overloaded or not using the GPU properly. You can check task manager to see if the GPU is being used when speaking to the mic (you should see spikes of usage when speaking). Also, you didn't add which zip file have you downloaded so I am not sure exactly which version are you using. So, please, write here the full name of the zip that you downloaded.

**Reply by veryangryapple** [reply]
MMVCServerSIO_win_onnxdirectML-cuda_v.1.5.3.18a is the first time I tried same thing heppend then I thought maybe I downloaded a wrong one then wanted to download a diffrent one so the second one was MMVCServerSIO_win_onnxgpu-cuda_v.1.5.3.18a

I'll try the advices you gave me when I am not busy thank you for all the help you gave me so far

**Comment by AnnoyingFox** [comment]
Yes, for nvidia GPU use MMVCServerSIO_win_onnxgpu-cuda_v.1.5.3.18a

**Reply by veryangryapple** [reply]
it works flawlessly after updating the drivers and switching to VAC instead of VB thank you

**Reply by AnnoyingFox** [reply]
yay that's great!
